subsys/net/ip/l2/gptp/gptp_user_api.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_DEBUG_GPTP)
+#define SYS_LOG_DOMAIN "net/gptp"
+#endif
+
+#include <ptp_clock.h>
+#include <net/gptp.h>
+
+#include "gptp_messages.h"
+#include "gptp_data_set.h"
+
+#include "../../net_private.h"
+
+static sys_slist_t phase_dis_callbacks;
+
+void gptp_register_phase_dis_cb(struct gptp_phase_dis_cb *phase_dis,
+		gptp_phase_dis_callback_t cb)
+{
+	sys_slist_find_and_remove(&phase_dis_callbacks, &phase_dis->node);
```

Comment:
the callback could take benefit of a "previous" parameter, so you could call sys_slist_remove(), much faster. Morever that you are running these in a SYS_SLIST_FOR_EACH_NODE_SAFE() below.

Reply:
Not worth optimizing the register function, it is called typically only once for each application.